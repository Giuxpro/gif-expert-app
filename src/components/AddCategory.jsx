import PropTypes from 'prop-types'
import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const onAddInputValue = (e) => {
    setInputValue(e.target.value);
  };

  const onHandleSubmit = (e) => {
    e.preventDefault();
    if( inputValue.trim().length <= 1) return;
    onNewCategory(inputValue);
    setInputValue('');
  };

 
  return (
    <form onSubmit={onHandleSubmit} aria-label="form">
      <input 
          type="text" 
          placeholder="Buscar gifs"
          value={ inputValue }
          onChange={onAddInputValue}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}