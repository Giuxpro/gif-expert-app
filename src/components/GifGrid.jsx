import PropTypes from 'prop-types';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs.js';



export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );

  
  return (
    <>  
        {
            isLoading && ( <h2>Cargando...</h2>)
        }
        <h3>{category}</h3>
        <div className='card-grid'>
            {
                images.map((images) =>(
                    <GifItem key={images.id} {...images} />
                ))
            }
        </div>
    </>
  )
}

GifGrid.propTypes = {
    category: PropTypes.string,
}