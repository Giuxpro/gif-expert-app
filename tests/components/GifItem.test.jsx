import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en < GifItem/>', () => {
  const title = 'titulo'
  const url = 'https://one-punch.com/saitama.jpg/'
  test('It should pass the snapshot', () => {
      const { container } = render( <GifItem title={title} url={url}/>)
      expect(container).toMatchSnapshot();
  });

  test('It should show the correct img with the url and alt', () => {
      render( <GifItem title={title} url={url}/>)
      // screen.debug()
      // expect(screen.getByRole('img').src).toBe(url);
      // expect(screen.getByRole('img').alt).toBe(title);
      const { src, alt } = screen.getByRole('img');
      expect( src ).toBe( url );
      expect( alt ).toBe( title );
  });

  test('It should return the title on the component', () => {
    render(<GifItem title={title} url={url}/>)
    expect( screen.getByText( title )).toBeTruthy();
  });
});