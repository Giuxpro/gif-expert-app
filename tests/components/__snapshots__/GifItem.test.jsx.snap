// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en < GifItem/> It should pass the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://one-punch.com/saitama.jpg/"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
